### DISCUSSION TITLE
[Angular Universal] 使用 Prerender 建立自己的 Static Site Generator | 全端開發人員天梯

### DISCUSSION THREAD

**Discussion by giscus** [wellwind/wellwind.github.io#42] [Announcements]
# [Angular Universal] 使用 Prerender 建立自己的 Static Site Generator | 全端開發人員天梯

隨著 Angular 不斷的改版，原來難用且功能差強人意的 Angular Universal 在不知不覺已經變得相當完整及強大，搭配 Prerender 功能，不用再依靠如 Hexo、Scully 等工具，要刻出屬於自己的 Static Site Generator (靜態網站產生器) 也變得相當容易，今天就來簡單看一下 Angular Universal 現在變得多麼簡單，且透過 Prerender 功能來打造一個簡單的 Static Site Generator 吧！

https://fullstackladder.dev/blog/2021/10/16/static-site-generator-using-angular-universal-prerender

**Comment by hochun836** [comment]
在網路上會看到一些名詞 : CSR, SSR, SSG, ISR
想請問一下，SSG 和 ISR 都是屬於 Prerender 的一種嗎 ?

**Reply by wellwind** [reply]
我認為都算是喔，都是屬於 Prerender 技術的實現

**Comment by hochun836** [comment]
想再請問一下
最近自己也想用 Angular 重新建置自己的部落格
像是大大的部落格，文章統計 / 分類統計 / 標籤統計 / 全站搜尋
這四大功能是怎麼實現的 🤔

**Reply by wellwind** [reply]
我有先把一些站台的摘要資訊整理起來，因為是靜態的，因此每次 build 時都會自動產生，之後抓這個檔案就好了

**Reply by hochun836** [reply]
原來如此 🙏 感謝
那想請問"搜尋" 🤔
也是事先準備好ㄧ份資料供畫面搜尋嗎

**Reply by wellwind** [reply]
是的，所以我的搜尋是找不到內文的 XD
